framework/src/org/checkerframework/framework/type/BoundsInitializer.java

```diff
@@ -391,36 +456,62 @@ public void initializeExtendsBound(final AnnotatedWildcardType wildcard) {
-                    final List<? extends TypeMirror> actualTypeArgs = actualType.getTypeArguments();
-                    for (int i = 0; i < actualTypeArgs.size(); i++) {
-                        final AnnotatedTypeMirror annoTypeArg =
-                                AnnotatedTypeMirror.createType(
-                                        actualTypeArgs.get(i), declaredType.atypeFactory, false);
-
-                        final BoundPathNode node = addPathNode(new TypeArgNode(i));
-                        typeArgs.add(replaceOrVisit(annoTypeArg));
-                        removePathNode(node);
+                typeArgs = new ArrayList<>();
+                for (int i = 0; i < typeElement.getTypeParameters().size(); i++) {
+                    TypeMirror javaTypeArg =
+                            getJavaType(declaredType, typeElement.getTypeParameters(), i);
+                    final AnnotatedTypeMirror atmArg =
+                            AnnotatedTypeMirror.createType(
+                                    javaTypeArg, declaredType.atypeFactory, false);
+                    typeArgs.add(atmArg);
+                    if (atmArg.getKind() == TypeKind.WILDCARD && declaredType.wasRaw()) {
+                        ((AnnotatedWildcardType) atmArg).setUninferredTypeArgument();
                     }
                 }
-                declaredType.setTypeArguments(typeArgs);
             } else {
+                typeArgs = declaredType.typeArgs;
+            }
 
-                final List<AnnotatedTypeMirror> typeArgs =
-                        new ArrayList<>(declaredType.getTypeArguments());
-                final List<AnnotatedTypeMirror> typeArgReplacements =
-                        new ArrayList<>(typeArgs.size());
-                for (int i = 0; i < typeArgs.size(); i++) {
-                    final AnnotatedTypeMirror typeArg = typeArgs.get(i);
-                    final BoundPathNode node = addPathNode(new TypeArgNode(i));
-                    typeArgReplacements.add(replaceOrVisit(typeArg));
-                    removePathNode(node);
+            List<AnnotatedTypeMirror> typeArgReplacements = new ArrayList<>(typeArgs.size());
+            for (int i = 0; i < typeArgs.size(); i++) {
+                AnnotatedTypeMirror typeArg = typeArgs.get(i);
+                BoundPathNode node = addPathNode(new TypeArgNode(i));
+                if (typeArg.getKind() == TypeKind.WILDCARD) {
+                    ((AnnotatedWildcardType) typeArg)
+                            .setTypeVariable(typeElement.getTypeParameters().get(i));
                 }
+                typeArgReplacements.add(replaceOrVisit(typeArg));
+                removePathNode(node);
+            }
+
+            declaredType.setTypeArguments(typeArgReplacements);
+        }
 
-                declaredType.setTypeArguments(typeArgReplacements);
+        Map<TypeVariable, WildcardType> rawTypeWildcards = new HashMap<>();
```

Comment:
Can you add some documentation for this field, please.
Why is the field default visibility?